treasury-factory/src/lib.rs

```diff
@@ -52,14 +55,15 @@ impl Contract {
                     "new_account_id": new_instance_contract_id.clone(),
                     "options": {
                         "full_access_keys": [env::signer_account_pk(),admin_full_access_public_key],
-                        "contract_bytes_base64": include_str!("../treasury_web4.wasm.base64.txt")
+                        // This is the base64 encoded wasm binary of the WebAssembly Text format file here: https://github.com/petersalomonsen/quickjs-rust-near/blob/main/factory/min_self_upgrade_contract.wat
+                        "contract_bytes_base64": "AGFzbQEAAAABHgZgAX4AYAJ+fgBgAX4BfmACfn4BfmADfn5+AGAAAAKTAQYDZW52EmN1cnJlbnRfYWNjb3VudF9pZAAAA2VudgVpbnB1dAAAA2Vudg1yZWFkX3JlZ2lzdGVyAAEDZW52DHJlZ2lzdGVyX2xlbgACA2VudhRwcm9taXNlX2JhdGNoX2NyZWF0ZQADA2VudiRwcm9taXNlX2JhdGNoX2FjdGlvbl9kZXBsb3lfY29udHJhY3QABAMCAQUFAwEAIAcLAQd1cGdyYWRlAAYKMgEwAQF+QgAQAEIAQoAIEAJCABADQoAIEAQhAEIAEAFCAEKAEBACIABCABADQoAQEAUL"
```

Comment:
I don't want to keep this door open for potential front-running receipts that call `upgrade` with an arbitrary contract during the treasury deployment. We either need to only allow the `upgrade` calls by this factory only, or deploy web4 contract as we did before. Can we still do that or is it too big now?

Reply:
It is still below the limit, but it is growing. The motivation for changing this now, is because of the view method that provides the web4 wasm binary to the self_upgrade function, where it is more future-proof to provide the wasm binary than a base64 encoded one. And since the web4 contract binary is stored inside the treasury-factory contract wasm, we don't want both a binary and base64 encoded version. I think then a better option is to have a check in the mini-self-upgrade contract, to make sure that it is only treasury-factory.near that can update it. Does that sound ok?

Reply:
> I think then a better option is to have a check in the mini-self-upgrade contract, to make sure that it is only treasury-factory.near that can update it. Does that sound ok?

The problem is that it won't work for testnet or staging testing if we hardcode the account name, and making it configurable might be cumbersome - patching the binary blob with the treasury account id is a bit scary.

Reply:
suggestion: we can modify the binary blob with the current account id at runtime. We can make a placeholder for an account id in the data section of the mini-self-upgrade wasm, and fill that with the current account id before encoding it as base64 and sending it to the linkdrop contract

Reply:
This works now @frol . It would of course be event better if the linkdrop contract supported receiving wasm binaries without encoding as base64 or in json, but for now this solution should work.